workflow-controller/controller-service/src/main/java/io/hyscale/controller/util/ServiceProfileUtil.java

```diff
@@ -17,84 +17,195 @@
+        }
+        
+        List<String> servicesWithoutProfile = services.stream().filter(each -> !serviceVsProfile.containsKey(each))
+                .collect(Collectors.toList());
+
+        if (!servicesWithoutProfile.isEmpty() || !servicesWithMultipleProfile.isEmpty()) {
+            String errorMessage = getErrorMessage(servicesWithoutProfile, servicesWithMultipleProfile);
+            WorkflowLogger.persist(ControllerActivity.ERROR_WHILE_PROCESSING_PROFILE, LoggerTags.ERROR, profileName, errorMessage);
+            HyscaleException hyscaleException = new HyscaleException(ControllerErrorCodes.ERROR_WHILE_PROCESSING_PROFILE,
+                    ToolConstants.INVALID_INPUT_ERROR_CODE, errorMessage);
+            logger.error(hyscaleException.getMessage());
+            throw hyscaleException;
+        }
+        
+        return serviceVsProfile.entrySet().stream().map(each -> each.getValue()).collect(Collectors.toList());
+    }
+    
+    private static String getErrorMessage(List<String> servicesWithoutProfile,
+            List<String> servicesWithMultipleProfile) {
+        StringBuilder errMsg = new StringBuilder();
+
+        if (!servicesWithoutProfile.isEmpty()) {
+            errMsg.append(NO_PROFILE_FOR_SERVICES);
+            servicesWithoutProfile.forEach(each -> errMsg.append(each).append(", "));
+        }
+        if (!servicesWithMultipleProfile.isEmpty()) {
+            errMsg.append(MULTIPLE_PROFILES_FOR_SERVICES);
+            servicesWithMultipleProfile.forEach(each -> errMsg.append(each).append(", "));
         }
-        return serviceProfileMap;
+        return HyscaleStringUtil.removeSuffixStr(errMsg, ", ");
     }
 
     /**
      * Gets service name from profile file.
      * 1.returns service name if present with the key {@link HyscaleSpecFields#overrides}.
-     * 2.else returns null when file is null or throws relative HyscaleException.
+     * 2.else returns null when file is null or throws relevant HyscaleException.
      *
      * @param profileFile
      * @return service name
      * @throws HyscaleException
      */
     public static String getServiceNameFromProfile(File profileFile) throws HyscaleException {
-       return get(profileFile,HyscaleSpecFields.overrides);
+        return get(profileFile, HyscaleSpecFields.overrides);
     }
 
     /**
      * Gets profile name from profile file.
      * 1.returns profile name if present with the key @HyscaleSpecFields#environment.
-     * 2.else returns null when file is null or throws relative HyscaleException.
+     * 2.else returns null when file is null or throws relevant HyscaleException.
      *
      * @param profileFile
      * @return profile or environment name
      * @throws HyscaleException
      */
-    public static String getProfileName(File profileFile) throws HyscaleException{
-       return get(profileFile,HyscaleSpecFields.environment);
+    public static String getProfileName(File profileFile) throws HyscaleException {
```

Comment:
Why do you still need this ?

Reply:
To validate profile file name and display warning
To create dependency between service and profile